pkg/storegateway/indexheader/reader_pool_test.go

```diff
@@ -28,54 +28,92 @@ import (
 
 func TestReaderPool_NewBinaryReader(t *testing.T) {
 	tests := map[string]struct {
-		lazyReaderEnabled     bool
-		lazyReaderIdleTimeout time.Duration
+		lazyReaderEnabled             bool
+		lazyReaderIdleTimeout         time.Duration
+		eagerLoadReaderEnabled        bool
+		persistLazyLoadedHeaderFn     func(blockId ulid.ULID) lazyLoadedHeadersSnapshot
+		expectedLoadCountMetricBefore int
+		expectedLoadCountMetricAfter  int
```

Comment:
It's not immediately clear what "before" and "after" here are relative to - might be worth clarifying the names.